src/query/query_eval.rs

```diff
@@ -0,0 +1,76 @@
+use crate::index::Index;
+use crate::metadata::Packet;
+use crate::query::query_types::*;
+use crate::query::QueryError;
+
+pub fn eval_query(index: Index, query: QueryNode) -> Result<Vec<Packet>, QueryError> {
+    match query {
+        QueryNode::Latest(inner) => eval_latest(index, inner),
+        QueryNode::Lookup(field, value) => eval_lookup(index, field, value),
+    }
+}
+
+fn eval_latest(index: Index, inner: Option<Box<QueryNode>>) -> Result<Vec<Packet>, QueryError> {
+    match inner {
+        Some(inner_node) => Ok(vec![eval_query(index, *inner_node)?
+            .last()
+            .unwrap()
```

Comment:
There was a hint in the rust book about unwrap with a self-documenting reason on failure